### DISCUSSION TITLE
How do I train Ham/Spam?

### DISCUSSION THREAD

**Discussion by blakegonzales** [docker-mailserver/docker-mailserver#3609] [Q&A] [answered]
First off, let me apologize if this is covered elsewhere.  I've looked through the discussions and the website and didn't find much.

I'm using rspamd with DMS and would like to train several dozen ham and spam emails.  There is some documentation here: https://rspamd.com/doc/quickstart.html#using-rspamc-console-routine

`#Training the Bayesian classifier`
`rspamc learn_spam < file.eml`
`rspamc learn_ham file.eml`

Although, I cannot seem to find `rspamc` 🤨.  Maybe there is a better method?

I would appreciate someone pointing me in the right direction!

**Comment by polarathene** [comment]
@georglauterbach may be able to assist when they have time.

Some of the rspamd integration requires using DMS specific configuration layer AFAIK, so this may require an update to our docs if not covered.

**Comment by georglauterbach** [comment, marked as answer]
> First off, let me apologize if this is covered elsewhere. I've looked through the discussions and the website and didn't find much.
> 
> I'm using rspamd with DMS and would like to train several dozen ham and spam emails. There is some documentation here: https://rspamd.com/doc/quickstart.html#using-rspamc-console-routine
> 
> `#Training the Bayesian classifier` `rspamc learn_spam < file.eml` `rspamc learn_ham file.eml`
> 
> Although, I cannot seem to find `rspamc` 🤨. Maybe there is a better method?

I am not completely sure I understand what you mean by that, but I will just take a guess: you don't know where to acquire the `rspamc` command? If this is the case, you'll find that the solution is very nice and simple: you do not need to do anything; `rspamc` is a command that is already installed in DMS (when you install Rspamd, the command `rspamc` comes as well), i.e. `rspamc` is in `$PATH`:

```console
$ docker exec -ti mailserver bash
root@mailserver # which rspamc
/usr/bin/rspamc
root@mailserver # # now you can use it
root@mailserver # rspamc learn_spam /path/to/spam/email
```

If I'm not mistaken, the results are persisted in Redis, which comes pre-configured as well if you didn't disable it manually.

**Comment by blakegonzales** [comment]
This is very helpful!  Thank you!!
